_posts/2016-09-29-Rust-1.12.md

```diff
@@ -0,0 +1,381 @@
+the compiler and less bloat at runtime. In addition, because MIR is a simpler
+'language' than the full AST, it's also easier to implement compiler passes on,
+and easier to verify that they are correct.
+
+The improvements in compile times of some workloads due to this change alone
+are significant:
+
+|               | 1.11   | 1.12  | compile time |
+|---------------|--------|-------|--------------|
+| inflate-0.1.0 | 10.17s | 4.32s | 2.35x faster |
+| hyper-0.5.0   | 24.03s | 5.23s | 4.58x faster |
+| helloworld    | 0.17s  | 0.14s | 1.20x faster |
+| regex 0.1.30  | 9.63s  | 2.39s | 4.03x faster |
+
+It is worth remembering that not all workloads may see this large of an
+improvement: it depends on a large number of factors.
+
+The compiler's [memory usage](https://en.wikipedia.org/wiki/Resident_set_size)
+while compiling these crates was also improved:
+
+|               | 1.11  | 1.12  | % of original size |
+|---------------|-------|-------|--------------------|
+| inflate-0.1.0 | 175MB | 135MB | 77.1%              |
+| hyper.0.5.0   | 321MB | 302MB | 94.1%              |
+| helloworld    | 127MB | 90MB  | 70.9%              |
+| regex.0.1.30  | 230MB | 194MB | 84.3%              |
+
+#### Other improvements
+
+* Many minor improvements to the documentation.
+* [`rustc` supports three new MUSL targets on ARM: `arm-unknown-linux-musleabi`,
+  `arm-unknown-linux-musleabihf`, and `armv7-unknown-linux-musleabihf`]
+  (https://github.com/rust-lang/rust/pull/35060).
+  These targets produce statically-linked binaries. There are no binary release
+  builds yet though.
+* In error descriptions,
+  [references](https://github.com/rust-lang/rust/pull/35611) and [unknown numeric
+  types](https://github.com/rust-lang/rust/pull/35080) have more human errors.
+* [The compiler can now be built against LLVM 3.9]
+  (https://github.com/rust-lang/rust/pull/35594)
+* [Test binaries now support a `--test-threads` argument to specify the number
+  of threads used to run tests, and which acts the same as the
+  `RUST_TEST_THREADS` environment variable]
+  (https://github.com/rust-lang/rust/pull/35414)  
+* [The test runner now emits a warning when tests run over 60 seconds]
+  (https://github.com/rust-lang/rust/pull/35405)
+* [Rust releases now come with source packages that can be installed by rustup
+  via `rustup component add rust-src`]
+  (https://github.com/rust-lang/rust/pull/34366).
+  The resulting source code can be used by tools and IDES, located in the
+  sysroot under `lib/rustlib/src`.
+
+See the [detailed release notes][notes] for more.
+
+#### Library stabilizations
+
+This release sees a number of small quality of life improvements for various
+types in the standard library:
+
+* [`Cell::as_ptr`](https://doc.rust-lang.org/std/cell/struct.Cell.html#method.as_ptr)
+  and
+  [`RefCell::as_ptr`](https://doc.rust-lang.org/std/cell/struct.RefCell.html#method.as_ptr)
+* `IpAddr`, `Ivp4Addr`, and `Ipv6Addr` have a few new methods.
```

Comment:
They're all separate, and there's a ton of them. Not sure how best to do the links here.